src/Core/Upgrade/UpgradeApiFilterVisitor.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Upgrade;
+
+use ApiPlatform\Core\Annotation\ApiFilter as LegacyApiFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\SearchFilter as LegacySearchFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\ExistsFilter as LegacyExistsFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\DateFilter as LegacyDateFilter;
+use ApiPlatform\Doctrine\Orm\Filter\SearchFilter;
+use ApiPlatform\Doctrine\Orm\Filter\ExistsFilter;
+use ApiPlatform\Doctrine\Orm\Filter\DateFilter;
+use ApiPlatform\Metadata\ApiFilter;
+use ApiPlatform\Metadata\Resource\DeprecationMetadataTrait;
+use Doctrine\Common\Annotations\AnnotationReader;
+use PhpParser\Node;
+use PhpParser\NodeVisitorAbstract;
+
+final class UpgradeApiFilterVisitor extends NodeVisitorAbstract
+{
+    use DeprecationMetadataTrait;
+    use RemoveAnnotationTrait;
+
+    private ?AnnotationReader $reader;
+    private \ReflectionClass $reflectionClass;
+
+    public function __construct(?AnnotationReader $reader, string $resourceClass)
+    {
+        $this->reader = $reader;
+        $this->reflectionClass = new \ReflectionClass($resourceClass);
+    }
+
+    /**
+     * @return int|Node|null
+     */
+    public function enterNode(Node $node)
+    {
+        if ($node instanceof Node\Stmt\Namespace_) {
+            $namespaces = [
+                ApiFilter::class,
+                SearchFilter::class,
+                ExistsFilter::class,
+                DateFilter::class,
```

Comment:
What about all the other filters?: https://github.com/api-platform/core/tree/2.7/src/Doctrine/Orm/Filter